File: src/main/java/com/gaye/dotoApp/dto/TacheResponse.java
package com.gaye.dotoApp.dto;

import java.time.LocalDateTime;

import com.gaye.dotoApp.model.ETAT;
import com.gaye.dotoApp.model.PRIORITY;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor
@Builder
@Data
public class TacheResponse {
    
    private Integer id;
    private String title;
    private LocalDateTime creatAt;
    private LocalDateTime lastModifiedAt;
    private String description;
    private ETAT etat;
    private PRIORITY priority;
    private LocalDateTime dueDate;
    private Integer projetId;
}

File: src/services/types/models/TacheResponse.ts
// TacheResponse.ts
// ✅ Type représentant les données reçues du backend pour une tâche

import { ETAT } from "../enum/ETAT";
import { PRIORITY } from "../enum/PRIORITY";

export interface TacheResponse {
  id: number; // Identifiant unique
  title: string; // Titre de la tâche
  creatAt: string; // Date de création (ISO string)
  lastModifiedAt: string; // Date de dernière modification (ISO string)
  description: string; // Description de la tâche
  etat: ETAT; // État de la tâche
  priority?: PRIORITY; // Priorité de la tâche (optionnelle)
  dueDate?: string; // Date d'échéance de la tâche (optionnelle)
  projetId: number; // ID du projet lié
}
